internal/registration/registration.go

```diff
@@ -125,8 +150,61 @@ func (r *Registration) registerDeviceWithRetries(interval int64) {
 	}
 }
 
+func (r *Registration) enrol() (bool, error) {
+	hardwareInformation, err := r.hardware.GetHardwareInformation()
+	if err != nil {
+		return false, err
+	}
+
+	enrolInfo := models.EnrolmentInfo{
+		Features: &models.EnrolmentInfoFeatures{
+			Hardware: hardwareInformation,
+		},
+		TargetNamespace: &r.targetNamepsace,
+	}
+
+	content, err := json.Marshal(enrolInfo)
+	if err != nil {
+		return false, err
+	}
+
+	data := &pb.Data{
+		MessageId: uuid.New().String(),
+		Content:   content,
+		Directive: "enrolment",
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second)
+	defer cancel()
+
+	// Call "Send" to send the enrolment message
+	response, err := r.dispatcherClient.Send(ctx, data)
+	if err != nil {
+		return false, err
+	}
+
+	parsedResponse, err := NewYGGDResponse(response.Response)
+	if err != nil {
+		return false, err
+	}
+
+	if parsedResponse.StatusCode == 208 {
+		return true, nil
+	}
+	return false, nil
+}
+
 func (r *Registration) registerDeviceOnce() error {
 
+	enrolled, err := r.enrol()
+	if err != nil {
+		return err
+	}
+
+	if !enrolled {
+		return fmt.Errorf("Device is not enrolled yet")
```

Comment:
Is this really an error? Do I understand correctly that it will end up here is the device is not approved yet?

Reply:
If we do not return an error, it'll not try to enrol again in the future.

Reply:
Is this visible in the logs?

Reply:
Yes: 
https://github.com/project-flotta/flotta-device-worker/blob/dc7ddd11603141944b4a4531c29b7afea9bb6731/internal/registration/registration.go#L106-L108

Reply:
What do you think about no logging it if this is expected error otherwise we could log it. It would be misleading to the user.

Reply:
at the same time is good that user knows that device is not enrolled yet (I used  for debug)

Reply:
I am not saying it is not needed but I am wondering whether error level is correct for this information